packages/interactivity/docs/2-api-reference.md

```diff
@@ -552,6 +552,78 @@ But it can also be used on other elements:
 
 When the list is re-rendered, the Interactivity API will match elements by their keys to determine if an item was added/removed/reordered. Elements without keys might be recreated unnecessarily.
 
+
+#### `wp-each`
+
+The `wp-each` directive is intended to render a list of elements. The directive can be used in `<template>` tags, being the value a path to an array stored in the global state or the context. The content inside the `<template>` tag is the template used to render each of the items.
+
+Each item is included in the context under the `item` name by default, so directives inside the template can access the current item.
+
+For example, let's consider the following HTML.
+
+```html
+<ul data-wp-context='{ "list": [ "hello", "hola", "olá" ] }'>
+  <template data-wp-each="context.list" >
+    <li data-wp-text="context.item"></li>
+  </template>
+</ul>
+```
+
+It would generate the following output:
+
+```html
+<ul data-wp-context='{ "list": [ "hello", "hola", "olá" ] }'>
+  <li data-wp-text="context.item">hello</li>
+  <li data-wp-text="context.item">hola</li>
+  <li data-wp-text="context.item">olá</li>
+</ul>
+```
+
+The prop that holds the item in the context can be changed by passing a suffix to the directive name. In the following example, we change the default prop `item` to `greeting`.
+
+```html
+<ul data-wp-context='{ "list": [ "hello", "hola", "olá" ] }'>
+  <template data-wp-each--greeting="context.list" >
```

Comment:
What is the use case to use this solution instead of
` <template data-wp-each="context.list" >
    <li data-wp-text="context.item"></li>`
, make it easier to read?

Reply:
Mostly for nested lists.

```js
store("myPlugin", {
  state: {
    names: ["Jon", "Aegon"],
    surnames: ["Snow", "Targaryen"],
    get fullname() {
      const { name, surname } = getContext();
      return `${name} ${surname}`;
    },
  },
});
```

```js
<ul>
  <template data-wp-each--name="state.names">
    <template data-wp-each--surname="state.surnames">
      <li data-wp-text="state.fullname"></li>
    </template>
  </template>
</ul>
```

This will render:

```js
<ul>
  <li>Jon Snow</li>
  <li>Jon Targaryen</li>
  <li>Aegon Snow</li>
  <li>Aegon Targaryen</li>
</ul>
```

Reply:
Cool. We should add this one or a similar example in the docs. But beware of spoilers! 😄

Reply:
> But beware of spoilers! 😄

Ups 😆